site/src/components/segmented-controls.md

```diff
@@ -0,0 +1,403 @@
+---
+title: Segmented Controls
+description: Segmented controls component in the Atlas Design System
+template: standard
+classType: Component
+classPrefixes:
+  - segmented-control
+  - segmented-controls
+  - segmented-controls-footer
+---
+
+## Segmented Control
+
+The segmented controls component is a tabs-like widget, containing a list of buttons for toggling an active view from a list. The view can be set by clicking the appropriate segmented control button, pressing the left or right arrow keys, or by clicking an optional previous or next button.
+
+```html
+<tab-container data-segmented-controls-container>
```

Comment:
I feel like we need some explanation around `tab-container` usage somewhere here. We haven't re-used any custom components before, so I see scenarios when people might get confused about these examples and not really understanding why it's here. @BenDMyers @homing1 @wibjorn any thoughts on this?

Reply:
I agree that adding further explanation would be good here. Such explanation should make it clear that `tab-container` is responsible for handling keyboard support and ARIA attributes out of the box.

I think this explanation should also link to `tab-container`'s documentation.

Reply:
Could this be non-blocking to this release?  Or would you prefer that I compile something asap?

Reply:
I really understand your eagerness to get this shipped (I've been in these shoes many times), but we have to make this right from the first round. As soon as it merged it becomes available to anyone and re-visiting later might cause issues in case component is already in use.

Also, the more I think about it - more I'm concerned about adding tab-container to Atlas. Atlas is a styling framework and introducing dependency on someone else's custom component we have no control over and adding extra JS to make it fit _our_ need bothers me. Should we move all the functionality logic to docs-ui as we normally do and have only styling in Atlas? Think about breadcrumbs component. it's very similar thing

@wibjorn @BenDMyers @homing1 please share your thoughts on this.

Reply:
I'm not necessarily against using GH's tab-container as a base for this. Atlas does have a minimal-ist approach to JS, which is to say we only add it when we need to ensure an interaction is usable and accessible - but prefer the CSS only approach first.

I'm more interested in the _how_ here, since that will inform us of the best approach.

Reply:
I think `tab-container` makes a good base for the logic, and I'd be generally in favor of continuing to use it. Accessible tabs will require JS -- CSS-only isn't sufficient -- and `tab-container` gives us most of what we need in a well-tested way.

My two gripes with `tab-container`, from having been involved in debugging the styles and refactoring the scripts are:

1. Many of its methods, which would have been useful for its previous/next button augmentation are instead private, which means some reverse engineering.
1. Its opinionated use of slots means markup inside generally has to be pretty specific, which is what's led to some of the specific nesting you see here. (Also, the use of shadow DOM has posed some annoying problems for our `outline-color` inheritance)

Reply:
My preference is to keep the JS here, since it's required for a usable experience. If we do run into major issues with private method, etc, then we can copy the code in an begin making our changes.